### ISSUE TITLE
`slice` should handle negative numbers

### ISSUE THREAD

**Issue by epage** [cobalt-org/liquid-rust#261]
Example test:
```rust
    assert_eq!(v!("ar"), filters!(slice, v!("foobar"), v!(-2), v!(2)));
```

**Comment by ryanyz10** [comment]
If this hasn't been fixed yet, I'd like to try fixing it. Is the relevant code in`src/filters/std/slice.rs`?

**Comment by epage** [comment]
Yes it is (sorry for the out of date link in the description).

Thanks for the help!  Whats interesting is it looks like we have some form of negative indexing support.  I do not remember why this was failing though.

**Comment by ryanyz10** [comment]
Yeah, it looks like reverse indexing should be handled. Why should the test fail if reverse indexing is handled?

**Comment by ryanyz10** [comment]
It looks like this test:
`assert_eq!(v!(""), filters!(Slice, v!("foobar"), v!(1), v!(0)));`
is failing, which I think is expected behavior since the slice length is 0.

**Comment by epage** [comment]
Weird, I wonder why I originally thought this was about negative numbers.

At least for the parts of liquid-rust I wrote, I wrote what I thought was the behavior.  I went back and ported integration tests over from the Ruby implementation which resulted in these issues being created.  The `conformance`  tests take precedence over anything that looks like intended behavior, even other tests.

Of course I could have made a mistake in porting the tests.  You can look at the file's history to when I checked in the ruby implementation or you can go look at [the current Ruby version](https://github.com/Shopify/liquid/blob/master/test/integration/standard_filter_test.rb#L77)

Again, thanks for looking into this!  You can also always reach me over on [gitter](https://gitter.im/cobalt-org/cobalt.rs) with questions.

**Comment by ryanyz10** [comment]
From the Ruby conformance tests, it looks like a slice of length 0 should just return an empty string/array. I changed the condition above to `length < 0` and the tests are passing now.
